runtime/lua/vim/diagnostic.lua

```diff
@@ -1822,10 +1822,11 @@ function M.open_float(opts, ...)
     -- Resolve options with user settings from vim.diagnostic.config
     -- Unlike the other decoration functions (e.g. set_virtual_text, set_signs, etc.) `open_float`
     -- does not have a dedicated table for configuration options; instead, the options are mixed in
-    -- with its `opts` table which also includes "keyword" parameters. So we create a dedicated
-    -- options table that inherits missing keys from the global configuration before resolving.
-    local t = global_diagnostic_options.float
-    local float_opts = vim.tbl_extend('keep', opts, type(t) == 'table' and t or {})
+    -- with its `opts` table. We create a dedicated options table (`float_opts`) that inherits
+    -- missing keys from the global configuration (`global_diagnostic_options.float`), which can
+    -- be a table or a function.
+    local t = global_diagnostic_options.float or global_diagnostic_options.float(0, bufnr)
```

Comment:
oh I may have made a mistake, should it be more like this?:

```suggestion
    local o = global_diagnostic_options
    local t = type(o.float) == 'table' and o.float or (type(o.float) == 'function' and o.float(0, bufnr) or {})
```